ci/filter-tasks.js

```diff
@@ -0,0 +1,156 @@
+// File to filter out tasks that don't need to be built
+var fs = require('fs');
+var path = require('path');
+var restClient = require('typed-rest-client/RestClient');
+var httpHandler = require('typed-rest-client/Handlers');
+var run = require('./ci-util').run;
+
+var makeOptionsPath = path.join(__dirname, '..', 'make-options.json');
+var makeOptions = JSON.parse(fs.readFileSync(makeOptionsPath).toString());
+
+var getTasksToBuildFromVersioning = async function() {
```

Comment:
Function name is confusing. Just getTasksToBuild? How we do that is abstracted.

Reply:
This is specifically to get the tasks to build for CI based on versioning info (as opposed to doing it based on PR info), but agreed this is still a pretty bad name. Maybe getCITasksToBuild? getTasksToBuildForCI? Naming functions has been a struggle for this entire PR :(